aptos-move/aptos-vm/src/move_vm_ext/respawned_session.rs

```diff
@@ -129,12 +130,15 @@ pub fn assert_layout_matches(
     layout_1: Option<&MoveTypeLayout>,
     layout_2: Option<&MoveTypeLayout>,
 ) -> Result<(), PanicError> {
-    //TODO[agg_v2](optimize): Don't compare the layouts everytime. Do this operation sporadically
-    if layout_1 != layout_2 {
-        return Err(code_invariant_error(format!(
-            "Layouts don't match when they are expected to: {:?} and {:?}",
-            layout_1, layout_2
-        )));
+    if layout_1.is_some() || layout_2.is_some() {
+        let mut rng = rand::thread_rng();
```

Comment:
Let's add a comment with a rationale why we do this sporadically?

Reply:
Done.